client/src/controllers/UnsavedController.ts

```diff
@@ -66,19 +66,24 @@ const DEFAULT_DURATIONS = {
  */
 export class UnsavedController extends Controller<HTMLFormElement> {
   static values = {
-    confirmation: { default: '', type: String },
+    confirmation: { default: false, type: Boolean },
     durations: { default: DEFAULT_DURATIONS, type: Object },
     force: { default: false, type: Boolean },
     hasComments: { default: false, type: Boolean },
     hasEdits: { default: false, type: Boolean },
     watch: { default: 'edits', type: String },
   };
 
-  /** Translated value for the beforeunload confirmation dialog, if empty no confirmation will show. */
-  declare confirmationValue: string;
+  /** Whether to show the browser confirmation dialog. */
+  declare confirmationValue: boolean;
   /** Configurable duration values. */
   declare durationsValue: typeof DEFAULT_DURATIONS;
-  /** When set to true, the form will always be considered dirty and the confirmation dialog will be forced to show. */
+  /** When set to true, the form will always be considered dirty.
+   * Useful for when the user just submitted an invalid form, in which case we
+   * consider the form to be dirty on initial load. Note that the
+   * `confirmationValue` must still be set to `true` in order for the browser
+   * confirmation dialog to appear.
+   */
```

Comment:
Clarify the docstring to avoid confusion like in https://github.com/wagtail/wagtail/pull/12099#discussion_r1682970073

Reply:
Thanks, I ended up moving this to the first commit.